packages/cli/lib/install.ts

```diff
@@ -0,0 +1,92 @@
+import { exec } from 'tinyexec';
+import { resolveCommand } from 'package-manager-detector';
+import type { Adder, Workspace, PackageManager, OptionValues, Question } from '@sveltejs/cli-core';
+import { createWorkspace } from '../commands/add/workspace.ts';
+import { createOrUpdateFiles } from '../commands/add/processor.ts';
+import { installPackages } from '../commands/add/utils.ts';
+
+type Addon = Adder<any>;
+export type InstallOptions<Addons extends AddonMap> = {
+	cwd: string;
+	addons: Addons;
+	options: OptionMap<Addons>;
+	packageManager: PackageManager;
+};
+
+export type AddonMap = Record<string, Addon>;
+export type OptionMap<Addons extends AddonMap> = {
+	[K in keyof Addons]: Partial<OptionValues<Addons[K]['options']>>;
+};
+
+export async function installAddon<Addons extends AddonMap>({
+	addons,
+	cwd,
+	options,
+	packageManager = 'npm'
+}: InstallOptions<Addons>): Promise<string[]> {
+	const filesToFormat = new Set<string>();
+
+	const mapped = Object.entries(addons).map(([, addon]) => addon);
+	const ordered = orderAddons(mapped);
+
+	for (const addon of ordered) {
+		const workspace = createWorkspace({ cwd, packageManager });
+		workspace.options = options[addon.id];
+
+		const files = await runAddon(workspace, addon);
+		files.forEach((f) => filesToFormat.add(f));
+	}
+
+	return Array.from(filesToFormat);
+}
+
+async function runAddon(
```

Comment:
nit: not sure these fits in a file called `install.ts`. Also it looks like this file is the only one in `lib` and has only to do with adders. What about merging this with `commands/add/utils.ts` or `utils/adders.ts`?

Reply:
this is no longer true in my opionion